### PR TITLE
Locking referrer bug 

### LINKED ISSUE #364: Locking referrer bug
Locking the referrer after POST doesn't work properly.

After the POST `_actionApply`, a `referrer_locked` is created. See below https://github.com/joomlatools/joomlatools-framework/blob/0624cc5c11398c7fd9c78351cac705413bbeed83/code/libraries/joomlatools/library/controller/behavior/editable.php#L171 but during `read` the lock can't be found in `$context->request->cookies` since the actual cookie isn't created yet in the browser until after it's been dispatched and loaded in the browser. See the following https://github.com/joomlatools/joomlatools-framework/blob/0624cc5c11398c7fd9c78351cac705413bbeed83/code/libraries/joomlatools/library/controller/behavior/editable.php#L116

This bug is consistent in all our extensions. The only reason why it redirects properly i.e. during `cancel` is because of this line https://github.com/joomlatools/joomlatools-framework/blob/0624cc5c11398c7fd9c78351cac705413bbeed83/code/libraries/joomlatools/library/controller/behavior/editable.php#L122
This line fails in TEXTman since a specific model state is set in the `pageable` model behavior to filter featured articles on `_beforeFetch` see below https://github.com/joomlatools/textman/blob/efb43d396de8de11b894baba2ce38b80a1a21c7b/code/site/components/com_textman/model/behavior/pageable.php#L48 which being added as a URL parameter in the current request URL and now differs to the referrer of the same page.

Regardless of the specific issue in TEXTman, the checking of the lock isn't working properly.

#### Steps to reproduce the behavior

1. Install an extension
1. Add a breakpoint in editable::setReferrer()'s `$context->request->cookies->has()` check
1. Run the debugger
1. Click Save `apply` on a record
1. Check the content of `$context->request->cookies->toArray()`

Current result: `referrer_locked` cookie does not exists thereby it proceeds in updating the `referrer` cookie.

### PR DESCRIPTION
**Pull request by ercanozkaya**
Closes #364

### CODE REVIEW

**Comment by ercanozkaya** [comment]
@jebbdomingo can you try this fix please and see if it fixes the issue?

**Comment by johanjanssens** [comment]
About using the session, good question. We could indeed do that, i think originally i wrote the code to support a not logged in scenario, but that is not something we ever used.

**Comment by ercanozkaya** [comment]
@jebbdomingo Got it. I think I fixed that case now. Can you take another look please?

**Comment by jebbdomingo** [comment]
@ercanozkaya working on my end

**Comment by ercanozkaya** [comment]
@johanjanssens could you do a final review please?

File: code/libraries/joomlatools/library/controller/behavior/editable.php [hunk 1]
```diff
@@ -179,7 +188,37 @@ protected function _lockReferrer(KControllerContextInterface $context)
...
+
+    /**
+     * Sets referrer lock for the request
+     *
+     * Dispatcher::_actionPost action calls Controller::_actionRender which in turn calls Controller::_actionRead
+     * before sending results. This leads to referrer being locked and unlocked in the same request.
+     *
+     * If referrer lock is set to true, we know that the lock happened in this request, so it is not unlocked yet.
+     *
+     * @param $locked
```
Comment (johanjanssens) [inline code/libraries/joomlatools/library/controller/behavior/editable.php:204]: @ercanozkaya Small type should  @param bool $locked

**Comment by johanjanssens** [comment]
@ercanozkaya Can you get this released, so @jebbdomingo  can release a TEXTman update?

**Pull request merged**